Sources/SwiftSyntaxMacrosTestSupport/Assertions.swift

```diff
@@ -34,41 +34,76 @@ public struct NoteSpec {
   public let column: Int
 
   /// The file and line at which this ``NoteSpec`` was created, so that assertion failures can be reported at its location.
-  internal let originatorFile: StaticString
+  internal let originatorFileID: String
+  internal let originatorFilePath: StaticString
   internal let originatorLine: UInt
+  internal let originatorColumn: UInt
```

Comment:
Instead of storing/passing all these 4 members every single time, what do you think about introducing a struct that holds them? That should also simplify the signature of `failureHandler` and make `failureHandler` extensible if we choose to add one of the parameters / deprecate one of them like `filePath`.